addons/calendar/models/calendar.py

```diff
@@ -1258,20 +1281,100 @@ def detach_recurring_event(self, values=None):
+            Special case: If we selected the first event of the recurrence we will return the real id because there is no need to split between future and past events.
+            (update future events from the first event = update all events)
+            :return: A new record set containing the recurrence for future events.
+        """
+        real_id, meeting_origin, data = self.get_origin_meeting_and_schedule_fields()
+
+        reccurent_dates = meeting_origin._get_recurrent_date_by_event()
+        # If we want all the future event from the first event, that means we want all events and there is no need to split it.
+        real_start = pytz.UTC.localize(fields.Datetime.from_string(data['start'])).astimezone(pytz.UTC)
+        try:
+            event_position = reccurent_dates.index(real_start)
+        except:
+            raise UserError('Save your recurrent settingq before choosing an action')
+        if event_position == 0:
+            return self.browse(real_id)
+
+        if data.get('rrule'):
+            # if use final date, split old reccurent event until self.start
+            if meeting_origin.end_type == 'end_date':
+                # The final date of the old event is equal to the start date of the new event.
+                data.update['final_date'] = meeting_origin.final_date
+                meeting_origin.final_date = data['start']
+            elif meeting_origin.end_type == 'count':
+                # If we use count, we should try to cut the count in two part.
+                old_count, new_count = self._compute_count_division(meeting_origin, reccurent_dates)
+                meeting_origin.count = old_count
+                data['count'] = new_count
+            data.update(
+                recurrent_id=real_id,
+                recurrent_id_date=data.get('start'),
+                rrule='',
+            )
+
+            # do not copy the id
+            data.pop('id')
+        return meeting_origin.copy(default=data)
+
+    @api.multi
+    def action_future_recurring_event(self):
+        """ Action used in the view to update all the future events from a selected date(a virtual event).
+            it will call the get_split_recurring_event method that will return a calendar event record that
+            has a recurrent settings for all future events.
+        """
+        meeting = self.get_split_recurring_event()
+        return {
+            'type': 'ir.actions.act_window',
+            'res_model': 'calendar.event',
+            'view_mode': 'form',
+            'res_id': meeting.id,
+            'target': 'current',
+            'flags': {'form': {'action_buttons': True}, 'initial_mode': 'edit'}
+        }
+
+    @api.multi
+    def action_all_recurring_event(self):
+        """ Action used in the view to update all the events of a recurrence. It will open the form view in edit mode
+            with the real calendar event id and not a virtual id.
+        """
```

Comment:
Used to update all occurences of a recurrent event. It'll open the form view of the real calendar event.